### PR TITLE
Add ActiveSupport instrumentation to Funky

### PR DESCRIPTION
**Pull request by dgb**
Adds support for ActiveSupport's instrumentation. Requests are
instrumented under the `request.funky` tag with uri, method, request and
response keys.

### CODE REVIEW

File: lib/funky/connections/base.rb [hunk 1]
```diff
@@ -5,14 +5,30 @@ class Base
...
+      def self.instrument(uri, request, &block)
+        data = {
+          request: request,
+          method: request.method,
+          request_uri: uri
+        }
+        if defined?(ActiveSupport::Notifications)
+          ActiveSupport::Notifications.instrument 'request.funky', data, &block
+        else
+          yield data
```
Comment (mkk-fullscreen) [inline lib/funky/connections/base.rb]: 🤔 why yield data here? it's going to blow up when you get down to `"Error #{response.code}: #{response.body}"`.

Comment (dgb) [reply lib/funky/connections/base.rb]: I could also just change it to make it more explicit. e.g. `block.call(data)`

**Comment by dgb** [comment]
@mkk-fullscreen updated with that more explicit call

File: lib/funky/connections/base.rb (hunk 1 above)
Comment (mkk-fullscreen) [reply lib/funky/connections/base.rb]: you wanna know what? i read `yield data` as `return data` rather than `block.call(data)`

:|

**Pull request merged**